### DISCUSSION TITLE
Automatically update some project fields, version uploading from GitHub

### DISCUSSION THREAD

**Discussion by Maelep** [modrinth/.github#5] [Feature requests]
### Is your feature request related to a problem? Please describe.

This could become a very time consuming and redundant problem when a developer has multiple mods.

### Describe the solution you'd like

I'm developing Rust game plug-ins and I really love your project but, from what I understand, you invite users to log in with their GitHub but don't integrate it to avoid redundancy and time consuming tasks.

It looks like that, when creating mods, you ask for things that are already done in our GitHub project (body, issue tracker, source code, wiki, ...). So, why not doing like https://umod.org : When adding a plug-in to uMod, it ask for the repository link then, when validating, connects a GitHub Applications to your repository that track your project. Then, every update done on GitHub is automatically updated on the website.

### Describe alternatives you've considered

The idea is making a GitHub Applications that will avoid redundancy by linking:
- Upload Version => GitHub Release
- README.md => Body
- Issue tracker => Project issues
- Source code => GitHub project
- Wiki => GitHub Wiki
- License => LICENSE
- And so on

![image](https://user-images.githubusercontent.com/53207714/148769226-1f0fad09-aade-4353-a0c4-44bc7ce9b770.png)

### Additional context

[https://umod.org](https://umod.org) has this feature for Rust game plug-ins.

**Comment by macbrayne** [comment]
How would dependencies work?

**Comment by Maelep** [comment]
Perhaps a file called `dependencies.json` containing a list of dependencies so the Modrinth could detect them.
If external to Modrinth show a link `[Plug-in Name](https://myplugin.com)` and if internal Modrinth could get more information.
```json
{
    "My Super Plug-in": "https://mySuperPlugin.com",
    "Another Super Plug-in": "https://modrinth.com/mod/anotherSuperPlugin"
}
```

**Comment by Minenash** [comment]
> Upload Version => GitHub Release

There's a lot more fields on versions then gh has, dependencies being the most complicated. The rest could probably be done by some text at the beginning or end of a release body.

**Comment by BradBot1** [comment]
Would love the README linking, I hate having to manually update my project on modrinth to mirror what changes have occured to my github

**Comment by Arbee4ever** [comment]
I made a [Cloudflare worker](https://github.com/Arbee4ever/mod_body_updater) that does exactly that. But if ur using that please change the webhook and github urls.

**Comment by BradBot1** [comment]
> I made a [Cloudflare worker](https://github.com/Arbee4ever/mod_body_updater) that does exactly that

Seems to be private and I feel like having this natively would just be better

**Comment by Arbee4ever** [comment]
Whoops, it's public now, and yes a native implementation would be a lot better, but my implementation works right now

**Comment by triphora** [comment]
Readme syncing is implemented in [minotaur](https://github.com/modrinth/minotaur) by the way

**Comment by BradBot1** [comment]
> Readme syncing is implemented in [minotaur](https://github.com/modrinth/minotaur) by the way

While that works for fabric and forge (since they utilise Gradle) it doesn't solve anything for Bukkit (most Bukkit plugins utilise maven) and since Modrinth supports plugins I don't think a "just use Gradle" approach is appropriate here

**Comment by Minenash** [comment]
And there's resource packs which don't tend to use either

**Comment by triphora** [comment]
Additional suggestions available at modrinth/labrinth#503

**Comment by funnyboy-roks** [comment]
I think that this feature would be super useful and would improve the usage of Modrinth quite a bit, however since it's not implemented, I created a GitHub action that automatically syncs for anyone who desires this functionality right now: [`modrinth-auto-description`](https://github.com/marketplace/actions/modrinth-auto-description)
